TableTopCampaignManager/DiscordBot.cs

```diff
@@ -49,5 +52,47 @@ public async Task InitializeCommands()
                 Console.WriteLine(json);
             }
         }
+        private async Task SlashCommandHandler(SocketSlashCommand command)
+        {
+            switch (command.Data.Name)
+            {
+                case "roll":
+                    await HandleRollCommand(command);
+                    break;  
+            }
+        }
+        private static int[] CalculateRoll(string userInput, SocketSlashCommand command)
+        {
+            Random random = new Random();
+            int numberOfRolls = Int32.Parse(userInput.Split('d')[0]);
+            int dieValue = Int32.Parse(userInput.Split('d')[1]);
+            int[] calculatedValues = new int[numberOfRolls];
+            if (numberOfRolls > 10)
+            {
+                command.RespondAsync(text: "That's more than 10 rolls! 10 is the maximum.");
+                return null;
+            }
+            else
+            {
+                for (int i = 0; i < numberOfRolls; i++)
+                {
+                    calculatedValues[i] = random.Next(1, dieValue + 1);
+                }
+                return calculatedValues;
+            }
+        }
+        private async Task HandleRollCommand(SocketSlashCommand command)
+        {
+            var userInput = CalculateRoll(command.Data.Options.First().Value.ToString(),command);
+            if (userInput != null)
```

Comment:
can you do if (userInput) in C#?

Reply:
I think I remember trying to do that and also `(userInput == True)` and for some reason it wasn't working. The function returns null if it fails so `(userInput != null)` is an appropriate way to write the `If` function.